server/workflow/workflow_server.go

```diff
@@ -590,8 +591,14 @@ func (s *workflowServer) getWorkflow(ctx context.Context, wfClient versioned.Int
 		return latest, nil
 	}
 	wf, err := wfClient.ArgoprojV1alpha1().Workflows(namespace).Get(ctx, name, options)
-	if err != nil {
-		return nil, sutils.ToStatusError(err, codes.Internal)
+	if wf == nil || err != nil {
```

Comment:
Do you think instead of merging two different functionality in one API, UI can make two API calls? So the API call will be very clear 1-on-1  functionality.

Reply:
The original purpose is to address use case mentioned in https://github.com/argoproj/argo-workflows/issues/10781 so that users only have to make one API call.